src/Kiota.Builder/Writers/HTTP/CodeClassDeclarationWriter.cs

```diff
@@ -0,0 +1,387 @@
+            { SecuritySchemeType.OpenIdConnect.ToString().ToLowerInvariant(), BearerAuth }
+        };
+    }
+
+    protected override void WriteTypeDeclaration(ClassDeclaration codeElement, LanguageWriter writer)
+    {
+        ArgumentNullException.ThrowIfNull(codeElement);
+        ArgumentNullException.ThrowIfNull(writer);
+
+        if (codeElement.Parent is CodeClass requestBuilderClass && requestBuilderClass.IsOfKind(CodeClassKind.RequestBuilder) && GetUrlTemplateProperty(requestBuilderClass) is CodeProperty urlTemplateProperty)
+        {
+            // Write short description
+            conventions.WriteShortDescription(requestBuilderClass, writer);
+            writer.WriteLine();
+
+            // Retrieve all query parameters
+            var queryParameters = GetAllQueryParameters(requestBuilderClass);
+
+            // Retrieve all path parameters
+            var pathParameters = GetPathParameters(requestBuilderClass);
+
+            var baseUrl = GetBaseUrl(requestBuilderClass);
+
+            // Write path parameters
+            WritePathParameters(pathParameters, writer);
+
+            // Write all query parameter variables
+            WriteQueryParameters(queryParameters, writer);
+
+            // Write all HTTP methods GET, POST, PUT, DELETE e.t.c
+            WriteHttpMethods(requestBuilderClass, writer, queryParameters, pathParameters, urlTemplateProperty, baseUrl);
+        }
+    }
+
+    /// <summary>
+    /// Retrieves all the query parameters for the given request builder class.
+    /// </summary>
+    /// <param name="requestBuilderClass">The request builder class containing the query parameters.</param>
+    /// <returns>An array of all query parameters.</returns>
+    private static CodeProperty[] GetAllQueryParameters(CodeClass requestBuilderClass)
+    {
+        // Retrieve all the query parameter classes
+        var queryParameterClasses = requestBuilderClass
+            .GetChildElements(true)
+            .OfType<CodeClass>()
+            .Where(static element => element.IsOfKind(CodeClassKind.QueryParameters))
+            .ToArray();
+
+        // Calculate the total number of query parameters
+        int totalQueryParameters = queryParameterClasses
+            .Sum(paramCodeClass => paramCodeClass.Properties
+            .Count(static property => property.IsOfKind(CodePropertyKind.QueryParameter)));
+
+        // Create an array to hold all query parameters
+        var queryParameters = new CodeProperty[totalQueryParameters];
+        int index = 0;
+
+        // Populate the array with query parameters
+        foreach (var paramCodeClass in queryParameterClasses)
+        {
+            foreach (var property in paramCodeClass.Properties
+                .Where(static property => property.IsOfKind(CodePropertyKind.QueryParameter)))
+            {
+                queryParameters[index++] = property;
+            }
+        }
+
+        return queryParameters;
```

Comment:
```suggestion
        return requestBuilderClass
            .GetChildElements(true)
            .OfType<CodeClass>()
            .Where(static element => element.IsOfKind(CodeClassKind.QueryParameters))
            .SelectMany(paramCodeClass => paramCodeClass.Properties)
            .Where(static property => property.IsOfKind(CodePropertyKind.QueryParameter))
            .ToArray();
```

This is going to result in:
- fewer iterations through the data
- fewer allocations
- better performance